### PR TITLE
Implement certifying and withdrawing Plutus script tests

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: cardano-cli/test/Test/Cli/JSON.hs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import qualified Data.Map.Strict as Map
+
+import           Cardano.CLI.Shelley.Run.Query
+
+import           Hedgehog (Property, checkSequential, discover, forAll, property, tripping)
+import           Hedgehog (Gen)
+import           Hedgehog.Gen as Gen
+import           Hedgehog.Range as Range
+
+-- TODO: Move to cardano-api
```
[the remark this answers is not in the document]
Comment (Jimbo4350) [reply cardano-cli/test/Test/Cli/JSON.hs:21]: Yep for a follow up PR

**Review by MarcFontaine — APPROVED** [review]
LGTM

**Review by deepfire — APPROVED** [review]

File: cardano-testnet/test/Main.hs [hunk 2]
```diff
@@ -24,8 +26,10 @@ tests :: IO TestTree
 tests = do
   pure $ T.testGroup "test/Spec.hs"
     [ T.testGroup "Spec"
-      [ H.testProperty "Spec.Plutus.Direct.TxInLockingPlutus" Spec.Plutus.Direct.TxInLockingPlutus.hprop_plutus
-      -- This hangs on Windows for an unknown reason
+      [ -- Fails to meet deadline on MacOS for an unknown reason
+        ignoreOnMac "Spec.Plutus.Direct.CertifyingAndWithdrawingPlutus" Spec.Plutus.Direct.CertifyingAndWithdrawingPlutus.hprop_plutus_certifying_withdrawing
```
Comment (newhoggy) [inline cardano-testnet/test/Main.hs:30]: 👍

**Review by newhoggy — APPROVED** [review]

**Pull request merged**